decode.go

```diff
@@ -570,20 +590,36 @@ func (u *unmarshaler) unmarshalMessage(node *yaml.Node, message proto.Message, f
 		return
 	}
 	// Decode the fields
-	fields := message.ProtoReflect().Descriptor().Fields()
+	msgDesc := message.ProtoReflect().Descriptor()
 	for i := 0; i < len(node.Content); i += 2 {
 		keyNode := node.Content[i]
-		if u.checkKind(keyNode, yaml.ScalarNode) {
-			if forAny && keyNode.Value == atTypeFieldName {
-				continue // Skip the @type field for Any messages
-			}
-			// Get the field Name, JSONName, or Number
-			if field := findField(keyNode.Value, fields); field != nil {
-				valueNode := node.Content[i+1]
-				u.unmarshalField(valueNode, field, message)
-			} else {
-				u.addErrorf(keyNode, "unknown field %#v, expended one of %v", keyNode.Value, getFieldNames(fields))
+		var key string
+		switch keyNode.Kind {
+		case yaml.ScalarNode:
+			key = keyNode.Value
+		case yaml.SequenceNode:
```

Comment:
What is this about? What does the YAML look like when there's a key that is not a scalar? I didn't even know that was allowed.

Reply:
yaml supports all the keys. This is here so a unquoted extention field is accepted

Reply:
Oh, I see -- so an unquoted extension field would look like an array with a single string element because of the enclosing brackets. I think that's worth a comment to clarify for readers.